synapse/lib/layer.py

```diff
@@ -3050,6 +3050,40 @@ def getTagPropValuCount(self, form, tag, prop, stortype, valu):
 
         return count
 
+    async def iterPropUniqValues(self, formname, propname, stortype):
+        try:
+            abrv = self.getPropAbrv(formname, propname)
+        except s_exc.NoSuchAbrv:
+            return
+
+        if stortype & 0x8000:
+            stortype = STOR_TYPE_MSGP
+
+        stor = self.stortypes[stortype]
+        abrvlen = len(abrv)
+
+        for lkey in self.layrslab.scanKeysByPref(abrv, db=self.byprop, nodup=True):
+
+            await asyncio.sleep(0)
```

Comment:
Is this a case where s_coro.pause() would be more performant?

Reply:
switched to pause